app/controllers/api/automate_workspaces_controller.rb

```diff
@@ -4,5 +4,32 @@ def edit_resource(type, id, data = {})
       obj = resource_search(id, type, collection_class(type))
       obj.merge_output!(data)
     end
+
+    def decrypt_resource(type, id = nil, data = nil)
+      raise BadRequestError, "Must specify an id for starting a #{type} resource" unless id
+
+      obj = resource_search(id, type, collection_class(type))
+      decrypt(obj, data)
+    end
+
+    def encrypt_resource(type, id = nil, data = nil)
+      raise BadRequestError, "Must specify an id for starting a #{type} resource" unless id
+
+      obj = resource_search(id, type, collection_class(type))
+      obj.encrypt(data['object'], data['attribute'], data['value'])
+      obj.reload
```

Comment:
Why does `obj` need to be reloaded? If `encrypt` is doing something to the object's data that isn't synced with its attributes, it seems like it should be `encrypt`'s responsiblity to maintain that consistency